Pidgin.Tests/ExpressionParserTestsFP.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Pidgin.Expression;
+using Xunit;
+using static Pidgin.Parser;
+using static Pidgin.Parser<char>;
+
+// Based off ExpressionParserTests
+
+namespace Pidgin.Tests
+{
+    public class ExpressionParserTestsFP : ParserTestBase
```

Comment:
These tests are more complex than they need to be - don't need any of the `Expression` machinery here. Just run various strings through the `Float` parser and check you get what you wanted back:

```csharp
[Fact]
public void TestFloat()
{
    AssertSuccess(Float.Parse("1.23"), 1.23);
    // etc
}
```

See the big `StringParserTests.cs` file for examples.

Reply:
Understood, will be resolved in the next upload.